src/Servers.jl

```diff
@@ -130,278 +109,196 @@ function check_rate_limit(tcp;
-
-Start a server listening on the provided `host:port`. `verbose` indicates whether server activity should be logged.
-Optional keyword arguments allow construction of `Server` on the fly if the `server` argument isn't provided directly.
-See `?HTTP.Server` for more details on server construction and supported keyword arguments.
-By default, `HTTP.serve` aims to "never die", catching and recovering from all internal errors. Two methods for stopping
-`HTTP.serve` include interrupting (ctrl/cmd+c) if blocking on the main task, or sending the kill signal via the server's in channel
-(`put!(server.in, HTTP.Servers.KILL)`).
-"""
-function serve end
-
-serve(host::IPAddr, port::Integer, args...; kwargs...) = serve(Sockets.InetAddr(host, port), args...; kwargs...)
-serve(host::AbstractString, port::Integer, args...; kwargs...) = serve(parse(IPAddr, host), port, args...; kwargs...)
-serve(host::AbstractString, args...; kwargs...) = serve(String(host), args...; kwargs...)
-function serve(host::Union{Sockets.InetAddr, String},
-               handler=req -> HTTP.Response(200, "Hello World!"),
-               logger::IO=stdout,
-               ;
-               verbose::Bool=true,
-               args...)
-    server = Server(handler, logger; cert="", key="", args...)
-    return serve(server, host, verbose)
-end
-serve(; host::IPAddr=Sockets.localhost,
-        port::Integer=8081,
-        handler=req -> HTTP.Response(200, "Hello World!"),
-        logger::IO=stdout,
-        args...) =
-    serve(host, port, handler, logger; args...)
+Base.isopen(s::Server2) = isopen(s.server)
+Base.close(s::Server2) = close(s.server)
+
+Sockets.accept(s::Server2{Nothing, S}) where {S} = Sockets.accept(s.server)
+Sockets.accept(s::Server2) = getsslcontext(accept(s.server), s.ssl)
 
 function getsslcontext(tcp, sslconfig)
-    ssl = SSLContext()
-    setup!(ssl, sslconfig)
-    associate!(ssl, tcp)
-    handshake!(ssl)
+    ssl = MbedTLS.SSLContext()
+    MbedTLS.setup!(ssl, sslconfig)
+    MbedTLS.associate!(ssl, tcp)
+    MbedTLS.handshake!(ssl)
     return ssl
 end
-getsslcontext(tcp, ::Nothing) = tcp
-
-const nolimit = typemax(Int)
 
 """
-    HTTP.listen([host=Sockets.localhost[, port=8081]]; <keyword arguments>) do http
+    HTTP.listen([host=Sockets.localhost[, port=8081]]; kw...) do req
         ...
     end
+    HTTP.listen(handler::HTTP.Handler, host=Sockets.localhost, port=8081; kw...)
 
-Listen for HTTP connections and execute the `do` function for each request.
+Listen for HTTP connections and either execute the `do` function for each request, or dispatch to the
+provided `handler`. Both the function or `handler` can accept an `HTTP.Request` object, or a raw
+`HTTP.Stream` connection to read & write from directly; for the latter, pass the `stream=true`
+keyword argument to operate on the `HTTP.Stream` directly.
 
 Optional keyword arguments:
- - `ssl::Bool = false`, use https.
- - `require_ssl_verification = true`, pass `MBEDTLS_SSL_VERIFY_REQUIRED` to
```

Comment:
`require_ssl_verification` : I'd rather not loose this option. I find it very convenient for doing quick local experiments without needing to set up certificates.

Reply:
All you have to do is `sslconfig=MbedTLS.SSLConfig(false)`; I think that might even be less characters than `require_ssl_verification=false`.

Reply:
`MbedTLS.SSLConfig` is implementation detail.
It will change when the TLS library changes.
What issue do you solve by removing this option ?